packages/@jsii/kernel/lib/kernel.ts

```diff
@@ -148,6 +136,25 @@ export class Kernel {
     };
   }
 
+  public invokeBinScript(
+    req: api.InvokeScriptRequest,
+  ): api.InvokeScriptResponse {
+    const result = cp.spawnSync(
+      'node',
+      [path.join(this._getPackageDir(req.pkgname), req.script)].concat(
+        req.args ?? [],
+      ),
+      { encoding: 'utf-8' },
+    );
+
+    return {
+      output: result.output,
+      stdout: result.stdout,
+      stderr: result.stderr,
+      status: result.status,
```

Comment:
This might be `null` if the app was killed by a signal... Might want to also pass `signal` though.
The contract for `spawnSync` guarantees that exactly one of `result.status` and `result.signal` will be non-`null`.

Following my suggestion for the type change above... This would be something like:
```suggestion
      status: result.status != null
        ? { code: result.status }
        : { signal: result.signal! }, // <- The `!` is to signal we "know" `result.signal` cannot be `null` here
```

Reply:
I do not think that it is a good idea to add as result one of property. It could be really inconvenient to parse such property.
What do you think if we add signal as additional property?